src/rez/package_cache.py

```diff
@@ -214,17 +230,43 @@ def add_variant(self, variant, force=False):
                     % package.repository
                 )
 
-        no_op_statuses = (
+        no_op_statuses = {
             self.VARIANT_FOUND,
-            self.VARIANT_COPYING,
-            self.VARIANT_COPY_STALLED
-        )
+            self.VARIANT_COPY_STALLED,
+        }
+        if not wait_for_copying:
+            # Copying variants are only no-ops if we want to ignore them.
+            no_op_statuses.add(self.VARIANT_COPYING)
 
         # variant already exists, or is being copied to cache by another thread/proc
         status, rootpath = self._get_cached_root(variant)
         if status in no_op_statuses:
+            if logger:
+                logger.warning(f"Not caching {variant.qualified_name}. "
+                               f"Variant {self.STATUS_DESCRIPTIONS[status]}")
             return (rootpath, status)
 
+        if wait_for_copying and status == self.VARIANT_COPYING:
+            spinner = PixelSpinner(f"Waiting for {variant.qualified_name} to finish copying. ")
+            while status == self.VARIANT_COPYING:
+                spinner.next()
+                time.sleep(self._COPYING_TIME_INC)
+                status, rootpath = self._get_cached_root(variant)
+            else:
+                # Status has changed, so report the change and return
+                if logger:
+                    if status in no_op_statuses:
+                        logger.warning(f"{variant.qualified_name} "
```

Comment:
I don't think it's something we want to tackle here, but it could be a surprising behavior that nothing is done to force a copy when a variant is stalled.

Reply:
No, I think the description of a stalled cache in the documentation makes it clear that it needs to be cleaned by hand in order to cache again in the future.  I don't think we want to change that behavior here.